ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/rollup/RollupCreatePayload.java

```diff
@@ -0,0 +1,197 @@
+import org.hyperledger.besu.ethereum.blockcreation.BlockTransactionSelector.TransactionValidationResult;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.encoding.TransactionDecoder;
+import org.hyperledger.besu.ethereum.rlp.RLPException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import io.vertx.core.Vertx;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RollupCreatePayload extends ExecutionEngineJsonRpcMethod {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RollupCreatePayload.class);
+  private final RollupMergeCoordinator mergeCoordinator;
+  private final BlockResultFactory blockResultFactory;
+
+  public RollupCreatePayload(
+      final Vertx vertx,
+      final ProtocolContext protocolContext,
+      final RollupMergeCoordinator mergeCoordinator,
+      final BlockResultFactory blockResultFactory) {
+    super(vertx, protocolContext);
+    this.mergeCoordinator = mergeCoordinator;
+    this.blockResultFactory = blockResultFactory;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ROLLUP_CREATE_PAYLOAD.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext) {
+    final Object requestId = requestContext.getRequest().getId();
+    final Hash parentBlockHash;
+    final List<Transaction> transactions;
+    final Address suggestedRecipient;
+    final long timestamp;
+    final Bytes32 prevRandao;
+    final List<?> rawTransactions;
+
+    try {
+      parentBlockHash = requestContext.getRequiredParameter(0, Hash.class);
+      rawTransactions = requestContext.getRequiredParameter(1, List.class);
+      prevRandao = requestContext.getRequiredParameter(2, Hash.class);
+      suggestedRecipient = requestContext.getRequiredParameter(3, Address.class);
+      timestamp = Long.decode(requestContext.getRequiredParameter(4, String.class));
+      transactions =
+          rawTransactions.stream()
+              .map(Object::toString)
+              .map(Bytes::fromHexString)
+              .map(TransactionDecoder::decodeOpaqueBytes)
+              .collect(Collectors.toList());
+    } catch (final RLPException | IllegalArgumentException e) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(), JsonRpcError.INVALID_PARAMS);
+    }
+    final Optional<BlockHeader> parentBlock =
+        protocolContext.getBlockchain().getBlockHeader(parentBlockHash);
+    if (parentBlock.isEmpty()) {
+      return replyWithStatus(requestId, RollupCreatePayloadStatus.INVALID_TERMINAL_BLOCK);
+    }
+
+    try {
+      final PayloadCreationResult result =
+          mergeCoordinator.createBlock(
```

Comment:
Could we replace this with a consensus agnostic way to create blocks?

Reply:
I don't get what you mean by "consensus agnostic way"? 
This method is meant to be used by the rollup consensus engine, like `engine_*` will be used by the ethereum consensus layer.